src/posix/platform/netif.cpp

```diff
@@ -523,6 +704,12 @@ void platformNetifStateChange(otInstance *aInstance, otChangedFlags aFlags)
     {
         UpdateLink(aInstance);
     }
+#if defined(__linux__)
+    if (OT_CHANGED_THREAD_NETDATA & aFlags)
+    {
+        UpdateExternalRoutes(aInstance);
```

Comment:
Can we apply the change after a small delay? (e.g. 1s). This can help us merge multiple Network Data changes to one update.

Reply:
According to an offline discussion, will leave it as is. Maybe we'll add OT timer API in the future so we can implement this mechanism.